src/algebra/Module/basic.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2019 Robert A. Spencer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robert A. Spencer
+-/
+import algebra.module
+import algebra.punit_instances
+import category_theory.types
+import linear_algebra.basic
+open category_theory
+
+universe u
+
+variables (R : Type u) [ring R]
+
+/-- An example of a module for any ring is the zero module -/
+instance module.zero_module : module R punit := module.of_core $
+  by refine {
+    smul := λ _ _, punit.star,
+    .. punit.comm_ring, ..};
+    intros; exact subsingleton.elim _ _
+
+/-- The category of R-modules and their morphisms. -/
+structure Module :=
+  (carrier : Type)
+  (prop_add_comm_group : add_comm_group carrier)
+  (prop_module : module R carrier)
+
+namespace Module
+  instance : has_coe_to_sort (Module R) :=
```

Comment:
We don't use additional indenting inside `namespace` blocks.

Reply:
Is there a good reason for this (and to indent sections)?  I often have to go scrolling through mathlib files to work out which namespace a lemma belongs to, and being able to fold on indents seems like a very easy solution.